core/pkg/telemetry/global.go

```diff
@@ -0,0 +1,150 @@
+package telemetry
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	prometheusclient "github.com/prometheus/client_golang/prometheus"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetricgrpc"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
+	"go.opentelemetry.io/otel/exporters/prometheus"
+	"go.opentelemetry.io/otel/metric/global"
+	"go.opentelemetry.io/otel/sdk/metric"
+	"go.opentelemetry.io/otel/sdk/resource"
+	"go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.18.0"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+const (
+	requestDurationName = "http_request_duration_seconds"
+	responseSizeName    = "http_response_size_bytes"
+	metricsExporterOtel = "otel"
+	exportInterval      = 2 * time.Second
+)
+
+// SetupGlobal setup global telemetry configurations based on provided configurations. This needs to be run before
+// obtaining any global.Meter or otel.Tracer based on global providers. Note that, metrics exporter works independent
+// of the collector target. However, overriding metrics exporter require collector target as we only support otel
+// collector
+func SetupGlobal(ctx context.Context, serviceName string, metricsExporter string, collectorTarget string) error {
```

Comment:
@thisthat  need your input on this approach of setting tracer and meter providers globally

Reply:
I don't like global objects since they introduce race conditions when executing tests in parallel (besides all other antipatterns). I suggest creating a file similar to what we have for metrics. Please also take a look at my comment on that file.